File: programmers/42578.js
function solution(clothes) {
  const closet = new Map();

  clothes.forEach(([_, type]) => {
    if (!closet.has(type)) closet.set(type, 0);

    closet.set(type, closet.get(type) + 1);
  });

  const ans = [...closet.values()].reduce((acc, cur) => acc * (cur + 1), 1) - 1;

  return ans;
}

File: programmers/42578.py
from typing import List
from functools import reduce


def solution(clothes: List[List[int]]) -> int:
    answer = 0
    cloth_dict = {}

    for [cloth, category] in clothes:
        cloth_dict.setdefault(category, [])
        cloth_dict[category].append(cloth)

    cloth_kinds = []
    for k in cloth_dict.keys():
        cloth_kinds.append(len(cloth_dict[k]) + 1)

    answer += reduce(lambda a, b: a * b, cloth_kinds) - 1
    return answer
